File: app/frontend/shared/domain/trigger.ts
export enum TriggerCompare {
  CONTAINS = 'in',
  PATTERN = 'pattern',
}

export enum TriggerFeature {
  TEXT_RESPONSE,
  AUDIO_RESPONSE,
}

export class Trigger {

  public readonly id: string
  public readonly title: string
  public readonly input: string
  public readonly compare: TriggerCompare
  public readonly discordServerId: string
  public readonly outputText: string|null

  constructor(values: {
    id: string,
    title: string,
    input: string,
    compare: string,
    outputText: string|null,
    discordServerId: string
  }) {
    this.title = values.title
    this.id = values.id
    this.input = values.input
    this.compare = values.compare as TriggerCompare
    this.discordServerId = values.discordServerId
    this.outputText = values.outputText
  }

}

export const triggerCompareOptions = () => Object.entries(TriggerCompare)

File: context/src/main/kotlin/com/usadapekora/context/domain/trigger/Trigger.kt
package com.usadapekora.context.domain.trigger

import com.usadapekora.context.domain.shared.Entity
import com.usadapekora.context.domain.shared.valueobject.UuidValueObject

data class Trigger(
    val id: TriggerId,
    var input: TriggerInput,
    var compare: TriggerCompare,
    var outputText: TriggerOutputText,
    var discordGuildId: TriggerDiscordGuildId
) : Entity() {
    data class TriggerId(override val value: String) : UuidValueObject(value)
    data class TriggerInput(val value: String)
    data class TriggerOutputText(val value: String?)
    data class TriggerDiscordGuildId(val value: String)
    enum class TriggerCompare(val value: String) {
        In("in"),
        Pattern("pattern");

        companion object {
            fun fromValue(value: String): TriggerCompare {
                val instance = values().find {
                    it.value == value
                } ?: throw TriggerException.InvalidValue()

                return instance
            }
        }

        override fun toString(): String = value
    }

    companion object {
        fun fromPrimitives(
            id: String,
            input: String,
            compare: String,
            outputText: String?,
            discordGuildId: String
        ) = Trigger(
            id = TriggerId(id),
            input = TriggerInput(input),
            compare = TriggerCompare.fromValue(compare),
            outputText = TriggerOutputText(outputText),
            discordGuildId = TriggerDiscordGuildId(discordGuildId)
        )
    }

    override fun id(): String = id.value
}
